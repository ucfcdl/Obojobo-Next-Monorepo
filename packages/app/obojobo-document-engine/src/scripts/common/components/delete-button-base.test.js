import React from 'react'
import DeleteButton from './delete-button'
import DeleteButtonBase from './delete-button-base'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

jest.mock('../page/focus')

describe('DeleteButton', () => {
	test('DeleteButton component', () => {
		const component = renderer.create(<DeleteButton />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('DeleteButton component with shouldPreventTab', () => {
		const component = renderer.create(<DeleteButton shouldPreventTab={true} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('DeleteButton component with tabIndex', () => {
		const component = renderer.create(<DeleteButton tabIndex={50} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('DeleteButton component disabled', () => {
		const component = renderer.create(<DeleteButton disabled />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('DeleteButton component with onClick', () => {
		const component = renderer.create(<DeleteButton onClick={() => {}} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('DeleteButton calls focus callback with ref argument', () => {
		const focus = require('../page/focus').default
		const wrapper = mount(<DeleteButton focus={focus} />)
		const inst = wrapper.find(DeleteButtonBase).instance()
		inst.focus()
		expect(focus).toHaveBeenCalledWith(inst.deleteButtonRef)
	})
})
