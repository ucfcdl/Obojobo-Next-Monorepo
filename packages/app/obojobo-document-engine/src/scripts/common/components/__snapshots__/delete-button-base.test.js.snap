// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton DeleteButton component 1`] = `
<div
  className="obojobo-draft--components--delete-button"
>
  <button>
    Delete
  </button>
</div>
`;

exports[`DeleteButton DeleteButton component disabled 1`] = `
<div
  className="obojobo-draft--components--delete-button"
>
  <button>
    Delete
  </button>
</div>
`;

exports[`DeleteButton DeleteButton component with onClick 1`] = `
<div
  className="obojobo-draft--components--delete-button"
>
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;

exports[`DeleteButton DeleteButton component with shouldPreventTab 1`] = `
<div
  className="obojobo-draft--components--delete-button"
>
  <button
    disabled={true}
    tabIndex="-1"
  >
    Delete
  </button>
</div>
`;

exports[`DeleteButton DeleteButton component with tabIndex 1`] = `
<div
  className="obojobo-draft--components--delete-button"
>
  <button
    tabIndex={50}
  >
    Delete
  </button>
</div>
`;
